src/Agent.js

```diff
@@ -137,6 +145,9 @@ export default class Agent {
 		if (query) {
 			req.query(query);
 		}
+		if (timeout!==undefined) {
+			req.timeout(timeout);
```

Comment:
What is the user-facing behavior of a timeout? Adding a timeout is a public interface change since it will affect the promise rejection. We need to document what will happen when there's a timeout compared to an HTTP error. This is more of a documentation comment.